usr/lib/common/mech_ssl3.c

```diff
@@ -920,12 +919,12 @@ static CK_RV ssl3_md5_only(STDLL_TokData_t *tokdata,
         TRACE_DEVEL("Digest Update failed.\n");
         return rc;
     }
-    len = sizeof(hash);
-    rc = digest_mgr_digest_final(tokdata, sess, FALSE, &digest_ctx, hash, &len);
+    len = MD5_HASH_SIZE;
+    rc = digest_mgr_digest_final(tokdata, sess, FALSE, &digest_ctx, outBuff,
+                                 &len);
 
-    if (rc == CKR_OK) {
+    if (rc != CKR_OK) {
         TRACE_DEVEL("Digest Final failed.\n");
-        memcpy(outBuff, hash, len);
```

Comment:
Ingo, please check the memcpy line you removed here. For the good case (rc == CKR_OK) it should be called.

Reply:
The old code did first receive the hash into variable hash, and then copy it as is to outBuff. The new code receives it directly into outBuff, so no memcpy needed anymore, regardless of good or bad case.

Reply:
ok - fine with me